pkg/git/git_integration_test.go

```diff
@@ -420,6 +442,17 @@ func TestFailureLatestPatchToPatchWrongBranch(t *testing.T) {
 	require.Equal(t, git.DiscoverResult{}, result)
 }
 
+func TestSuccessLatestPatchToLatest(t *testing.T) {
+	testRepo := newTestRepo(t)
+	defer testRepo.cleanup(t)
+
+	result, err := testRepo.sut.LatestPatchToLatest(testRepo.branchName)
+	require.Nil(t, err)
+	require.Equal(t, result.StartSHA(), testRepo.firstCommit)
```

Comment:
This is surprising to me. I expected it to use secondBranchCommit

Reply:
Yeah maybe the return value of `LatestTagForBranch` is wrong.

Reply:
So the issue appears to be that `TagsForBranch` orders the tags not by timestamp, but by tag value, ie: 0.1.1, 0.1.2, 1.17.0 even though they were committed in a different order. I'm not sure if this is an issue with the gogit library or if this is how a real git repo behaves.

Do you expect a scenario where the intermediate tags on a branch are of a completely different major/minor version?

Anyway, what I did was change the v0.1.2 tag to v1.17.1 so that the test branch contains both kinds of tags. Basically, the v0.1.2 tag gets ignored for the most part since it's last in the ordering, and now the test cases work as expected.

Reply:
Sounds good to me, thank you for the test-fix :)